pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -219,11 +240,23 @@ func (c *Controller) Reconcile(ps *model.PushContext) error {
 	if err != nil {
 		return fmt.Errorf("failed to list type Namespaces: %v", err)
 	}
-	namespaces := map[string]*corev1.Namespace{}
+	namespaces := make(map[string]*corev1.Namespace, len(nsl))
 	for _, ns := range nsl {
 		namespaces[ns.Name] = ns
 	}
+
+	secretList, err := c.secretLister.List(klabels.Everything())
```

Comment:
I'm not certain if we really need this, IIUC we should be able to end up with a correct status even if any secret is changed during reconciling? And SDS push does directly read secrets from the informer cache

Same for namespaces